src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -708,88 +705,282 @@ void Graph::ResolveComplexInplaceConflicts() {
     }
 }
 
-static inline bool isConstOutput(EdgePtr edge) {
-    return edge->getParent()->isConstant() && !edge->getChild()->isConstant();
+/**
+ * Partition the \clusters of Edges, by moving to the end and allocating at the same time
+ * the clusters which cannot be handled as part of generic memory solver algorithm.
+ * Such clusters meet one of the following criteria:
+ * - base edge of a cluster is already Allocated
+ * - base edge of a cluster is a "ov::element::string" type of edge
+ * - base edge of a cluster is a Constant edge
+ *
+ * @return a remaining number of clusters to process (left partition)
+ */
+static size_t AllocateStringsAndConstants(EdgeClusters& clusters,
+                                          const GraphContext::CPtr context) {
+    auto allocateConstantEdge = [context](const EdgePtr& edge) {
+        if (edge->getParent()->getType() == Type::Input) {
+            auto constNode = std::static_pointer_cast<node::Input>(edge->getParent());
+            edge->reuse(std::const_pointer_cast<IMemory>(constNode->getMemoryPtr()));
+        } else {
+            edge->externalAllocate(context->getWeightsCache());
+        }
+    };
+
+    auto allocateStringMemory = [context](const EdgePtr& edge) {
+        auto memory = std::make_shared<StringMemory>(context->getEngine(), edge->getDesc());
+        edge->reuse(memory);
+        return memory->getStringMemoryBlockPtr();
+    };
+
+    auto notAllocatedPartitionEnd =
+        std::partition(clusters.begin(), clusters.end(),
+                       [&allocateStringMemory, &allocateConstantEdge, &context](const EdgeCluster& cluster) {
+                           if (cluster.empty()) return false;
+
+                           auto baseEdgeIt = std::find_if(cluster.begin(), cluster.end(), [](const EdgePtr& edge) {
+                               return one_of(edge->getStatus(), Edge::Status::Allocated, Edge::Status::NeedAllocation);
+                           });
+
+                           OPENVINO_ASSERT(baseEdgeIt != cluster.end(), "Unexpected cluster state");
+
+                           const auto& baseEdge = *baseEdgeIt;
+                           if (baseEdge->getStatus() == Edge::Status::Allocated) {
+                               return false;
+                           }
+
+                           // Skip if the baseEdge does not require allocation
+                           if (baseEdge->getStatus() != Edge::Status::NeedAllocation) {
+                               return true;
+                           }
```

Comment:
May be I missing something, but the base edge has either `Edge::Status::Allocated` or `Edge::Status::NeedAllocation` status, simply following the search condition of the `std::find_if` call above. If the edge is `Allocated`, lambda returns, so `baseEdge->getStatus() != Edge::Status::NeedAllocation` check looks useless as if it is true, then the base edge is `Allocaed` and lambda returns.